src/SimplCommerce.WebHost/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -226,6 +231,49 @@ public static IServiceCollection AddCustomizedDataStore(this IServiceCollection
             return services;
         }
 
+        /// <summary>
+        /// Registers core application services with the dependency injection container.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add services to.</param>
+        /// <param name="configuration">The application configuration manager containing settings and connection strings.</param>
+        /// <returns>The same service collection so that multiple calls can be chained.</returns>
+        public static IServiceCollection AddApplicationServices(this IServiceCollection services, ConfigurationManager configuration)
+        {
+            services.AddHttpClient();
+            services.AddTransient(typeof(IRepository<>), typeof(Repository<>));
+            services.AddTransient(typeof(IRepositoryWithTypedId<,>), typeof(RepositoryWithTypedId<,>));
+            services.AddScoped<ITagHelperComponent, LanguageDirectionTagHelperComponent>();
+            services.AddTransient<IRazorViewRenderer, RazorViewRenderer>();
+            services.AddAntiforgery(options => options.HeaderName = "X-XSRF-Token");
+            services.AddCloudscribePagination();
+            services.AddMediatR(cfg => cfg.RegisterServicesFromAssembly(typeof(Program).Assembly));
+            
+            return services;
+        }
+
+        /// <summary>
+        /// Discovers and configures all modules in the application by finding and initializing their module initializers.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add module services to.</param>
+        /// <returns>The same service collection so that multiple calls can be chained.</returns>
+        public static IServiceCollection ConfigureModules(this IServiceCollection services)
```

Comment:
I'm agree with this extension, please revert the others, coz the order sometimes is matter